multiqc/modules/sortmerna/sortmerna.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from SortMeRNA """
+
+from __future__ import print_function
+from collections import OrderedDict
+import json
+import logging
+import os
+import re
+
+from multiqc.modules.base_module import config, BaseMultiqcModule
+from multiqc import plots
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='SortMeRNA', anchor='sortmerna',
+        href='http://bioinfo.lifl.fr/RNA/sortmerna/',
+        info="is a program tool for filtering, mapping and OTU-picking NGS reads in metatranscriptomic and metagenomic data.")
+
+        # Parse logs
+        self.sortmerna = dict()
+        for f in self.find_log_files(config.sp['sortmerna'], filehandles=True):
+            self.parse_sortmerna(f)
+
+        if len(self.sortmerna) == 0:
+            log.debug("Could not find any SortMeRNA data in {}".format(config.analysis_dir))
+            raise UserWarning
+
+        log.info("Found {} logs".format(len(self.sortmerna)))
+        self.write_data_file(self.sortmerna, 'multiqc_sortmerna')
+        log.debug(self.sortmerna)
+        # Add rRNA rate to the general stats table
+        headers = OrderedDict()
+        headers['rRNA_pct'] = {
+            'title': '% rRNA',
+            'description': '% rRNA',
+            'max': 100,
+            'min': 0,
+            'suffix': '%',
+            'scale': 'OrRd',
+            'format': '{:.1f}%'
+        }
+        self.general_stats_addcols(self.sortmerna, headers)
+
+        # Make barplot
+        self.intro += self.sortmerna_overall_barplot()
+        self.intro += self.sortmerna_detailed_barplot()
+
+    def parse_sortmerna(self, f):
+        s_name = None
+        post_results_start = False
+        post_database_start = False
+        db_number = 0
+        err = False
+
+        for l in f["f"]:
+            if "Reads file" in l:
+                s_name = os.path.basename(l.split(" ")[-1]).strip().split(".")[0]
+                self.sortmerna[s_name] = dict()
+            if "Results:" in l and not post_results_start:
+                post_results_start = True
+            if not post_results_start:
+                continue
+            if post_results_start and not post_database_start:
```

Comment:
This could probably be generalised / simplified a little by using nested loops:

```python
if "Results:" in l:
    while "By database:" not in l:
        for l in f["f"]:
            key, val = l.strip().split('=', 1)
            key = key.strip().replace(' ', '_') 
            try:
                self.sortmerna[s_name][ key ] = int(val)
            except ValueError:
                rcount = val.split('(', 1)[0]
                self.sortmerna[s_name][key] = int(rcount)
                self.sortmerna[s_name]["{}_pct".format(key)] = float(rcount) / float(self.sortmerna[s_name]["Total_reads"]) * 100.0
if "By database:" in l:
    while l.strip() != '':
        for l in f["f"]:
            # ...
```
(only a suggestion)